thriftbp/server_middlewares.go

```diff
@@ -380,3 +383,50 @@ func RecoverPanic(name string, next thrift.TProcessorFunction) thrift.TProcessor
 		},
 	}
 }
+
+// PrometheusMetricMiddleware returns middleware to track Prometheus metrics specific to the Thrift service.
+func PrometheusMetricMiddleware(serviceSlug string) thrift.ProcessorMiddleware {
+	return func(method string, next thrift.TProcessorFunction) thrift.TProcessorFunction {
+		process := func(ctx context.Context, seqID int32, in, out thrift.TProtocol) (success bool, err thrift.TException) {
+			start := time.Now()
+			activeRequestLabels := prometheus.Labels{
+				"thrift_service": serviceSlug,
+				"thrift_method":  method,
+			}
+			activeRequests.With(activeRequestLabels).Inc()
+
+			defer func() {
+				var exceptionTypeLabel, baseplateStatusCode, baseplateStatus string
+				successLabel := strconv.FormatBool(err == nil)
+				if err != nil {
+					exceptionTypeLabel = strings.TrimPrefix(fmt.Sprintf("%T", err), "*")
```

Comment:
🔕 I _think_ the trimming of `*` could be a premature optimization, as we rarely use an error type both as reference and as pointer, but I don't feel strongly either way

(also if `*` in labels will cause troubles in prometheus then yea we absolutely need to do that)

Reply:
regexp is a good point, it's fine to keep it